src/lexer.jl

```diff
@@ -21,8 +24,8 @@ end
 ishex(c::Char) = isdigit(c) || ('a' <= c <= 'f') || ('A' <= c <= 'F')
 iswhitespace(c::Char) = Base.UTF8proc.isspace(c)
 
-type Lexer{IO_t <: IO, T <: AbstractToken}
-    io::IO_t
+type Lexer{T}
```

Comment:
Not really, they can be any `IO` type, so you could `tokenize(open("path"))`. for example.